packages/rsocket-core/tests/Fragmenter.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { fragment, isFragmentable } from "../src/Fragmenter";
+import { FrameTypes } from "@rsocket/rsocket-types";
+
+describe("isFragmentable", () => {
+  it("returns false when fragmentSize is 0", () => {
+    const actual = isFragmentable(undefined, 0, undefined);
+    const expected = false;
+    expect(actual).toBe(expected);
+  });
+
+  it("returns false when frame size is smaller than fragmentSize", () => {
+    const actual = isFragmentable(
+      {
+        data: Buffer.from([]),
+        metadata: Buffer.from([]),
+      },
+      1000,
+      FrameTypes.REQUEST_FNF
+    );
+    const expected = false;
+    expect(actual).toBe(expected);
+  });
+
+  it("returns true when frame size is larger than fragmentSize", () => {
+    const actual = isFragmentable(
+      {
+        data: Buffer.from("hello world"), // 11
+        metadata: Buffer.from("hello world"), // 11
+      },
+      10,
+      FrameTypes.REQUEST_CHANNEL // 3
+    );
+    const expected = true;
+    expect(actual).toBe(expected);
+  });
+});
+
+// TODO: seems to be some issues with `fragment` method. Is `fragment` fully implemented/exercised?
+//  - last frame seems to not be produced
+//  - NPE thrown on `data`
+//  - metadata fragmentation is reference data on line 63
+//    - `metadata = payload.data.slice(metadataPosition, nextMetadataPosition);`
+//  - passing undefined as `metadata` produces unexpected behavior
```

Comment:
@OlegDokuka is `fragment` considered to be fully implemented? I ran into some unexpected behavior when trying to test it. I tried to address some of the items but I'm not familiar with what is exactly expected as valid output.